src/node/rpc/member_frontend.h

```diff
@@ -144,9 +145,60 @@ namespace ccf
   DECLARE_JSON_TYPE(SetModule)
   DECLARE_JSON_REQUIRED_FIELDS(SetModule, name, module)
 
+  struct JsBundleEndpointMethod
+  {
+    std::string js_module;
+    std::string js_function;
+  };
+  DECLARE_JSON_TYPE(JsBundleEndpointMethod)
+  DECLARE_JSON_REQUIRED_FIELDS(JsBundleEndpointMethod, js_module, js_function)
+
+  using JsBundleEndpoint = std::map<std::string, JsBundleEndpointMethod>;
+
+  struct JsBundleMetadata
+  {
+    std::map<std::string, JsBundleEndpoint> endpoints;
+  };
+  DECLARE_JSON_TYPE(JsBundleMetadata)
+  DECLARE_JSON_REQUIRED_FIELDS(JsBundleMetadata, endpoints)
+
+  struct JsBundle
+  {
+    JsBundleMetadata metadata;
+    std::vector<SetModule> modules;
+  };
+  DECLARE_JSON_TYPE(JsBundle)
+  DECLARE_JSON_REQUIRED_FIELDS(JsBundle, metadata, modules)
+
+  struct DeployJsApp
+  {
+    std::string name;
+    JsBundle bundle;
+  };
+  DECLARE_JSON_TYPE(DeployJsApp)
+  DECLARE_JSON_REQUIRED_FIELDS(DeployJsApp, name, bundle)
+
   class MemberEndpoints : public CommonEndpointRegistry
   {
   private:
+    static bool startswith(const std::string& s, const std::string& prefix)
```

Comment:
If there's a better place for such utility functions, let me know.

Reply:
`ds/nonstd.h`